addons/web/static/tests/views/view_dialogs/export_data_dialog_tests.js

```diff
@@ -1253,4 +1253,38 @@ QUnit.module("ViewDialogs", (hooks) => {
             await click(target.querySelector(".o_list_export_xlsx"));
         }
     );
+
+    QUnit.test("Export dialog: disable button during export", async function (assert) {
+        await makeView({
+            serverData,
+            type: "list",
+            resModel: "partner",
+            arch: `
+                <tree><field name="foo"/></tree>`,
+            actionMenus: {},
+            mockRPC(route, args) {
+                if (route === "/web/export/formats") {
+                    return Promise.resolve([{ tag: "xls", label: "Excel" }]);
+                }
+                if (route === "/web/export/get_fields") {
+                    return Promise.resolve(fetchedFields.root);
+                }
+            },
+        });
+
+        mockDownload(() => {
+            return new Promise(r => setTimeout(r, 1000));
+        });
```

Comment:
Try to never await an arbitrary amount of time in a test.
This will slow down the test execution.

Instead import `makeDeferred` from `@web/../tests/helpers/utils`.

It allows you to create a resolvable promise.

Usage:
```js
const def = makeDeferred();
mockDownload(() => def);
// ...

def.resolve();
// assert your thing, maybe you need to await a tick if resolving the promise would result in a render cycle.
```

Reply:
The test should run in let's say 400ms instead of about 2400ms here.

Reply:
Thanks!